### ISSUE TITLE
Context access in rules (docs)

### ISSUE THREAD

**Issue by jprochazk** [jprochazk/garde#59]
It should be possible to access the context in attributes. This would allow users to dynamically specify the values on rules.

```rust
struct State {
    min_length: usize,
}

#[derive(garde::Validate)]
#[garde(context(State as ctx))]
struct Foo<'a> {
    #[garde(length(min = ctx.min_length))]
    v: &'a str,
}

let mut s = State { min_length: 0 };
assert!(Foo { v: "abcd" }.validate(&s).is_ok());
s.min_length = 10;
assert!(Foo { v: "abcd" }.validate(&s).is_err());
```

Implementation would be:
- Allow `expr` in values for every rule that should be able to access the context
- Add the `as <IDENT>` syntax to the `context` rule (it would be optional)
- Rename `__garde_ctx` to the `<IDENT>` specified in the `context` attribute if present, otherwise keep it as `__garde_ctx`

**Comment by jprochazk** [comment]
Also update documentation to mention you can already use `self` in certain rules. Extending all the rules that take values to accept any `Expr` would be great.

**Comment by jprochazk** [comment]
Implemented in [v0.14.1](https://github.com/jprochazk/garde/releases/tag/v0.14.1), but keeping this open for the documentation part.

**Comment by jprochazk** [comment]
Done in https://github.com/jprochazk/garde/commit/edc2dcd78b29ca0c70f240857ce89462820342f5
